src/main/java/com/percero/agents/auth/services/AuthService2.java

```diff
@@ -0,0 +1,691 @@
+package com.percero.agents.auth.services;
+
+import com.percero.agents.auth.hibernate.AssociationExample;
+import com.percero.agents.auth.hibernate.AuthHibernateUtils;
+import com.percero.agents.auth.hibernate.BaseDataObjectPropertySelector;
+import com.percero.agents.auth.vo.*;
+import com.percero.agents.sync.hibernate.SyncHibernateUtils;
+import com.percero.agents.sync.metadata.*;
+import com.percero.agents.sync.services.ISyncAgentService;
+import com.percero.framework.bl.IManifest;
+import com.percero.framework.bl.ManifestHelper;
+import com.percero.framework.vo.IPerceroObject;
+import org.apache.log4j.Logger;
+import org.hibernate.*;
+import org.hibernate.criterion.Restrictions;
+import org.hibernate.exception.LockAcquisitionException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.*;
+
+/**
+ * This class handles AuthenticationRequest type authentication
+ */
+@Component
+public class AuthService2 {
+
+    private static Logger logger = Logger.getLogger(AuthService2.class);
+
+    @Autowired
+    private AuthProviderRegistry authProviderRegistry;
+
+    @Autowired
+    SessionFactory sessionFactoryAuth;
+
+    @Autowired
+    SessionFactory appSessionFactory;
+
+    @Autowired
+    ISyncAgentService syncAgentService;
+
+    @Autowired
+    IManifest manifest;
+
+    public AuthenticationResponse authenticate(AuthenticationRequest request) throws IllegalArgumentException{
+        if(!authProviderRegistry.hasProvider(request.getAuthProvider()))
+            throw new IllegalArgumentException(request.getAuthProvider()+" auth provider not found");
+
+        AuthenticationResponse response = new AuthenticationResponse();
+
+        IAuthProvider provider = authProviderRegistry.getProvider(request.getAuthProvider());
+
+        ServiceUser serviceUser = provider.authenticate(request.getCredential());
+
+        // Login successful
+        if(serviceUser != null) {
+            logger.info(provider.getID() + " Authentication success");
```

Comment:
Let's use debug here.  With 15,000 concurrent users the logs will fill up fast! ;-)